en/dev_log/ulog_file_format.md

```diff
@@ -41,233 +48,331 @@ The file consists of three sections:
 ----------------------
 ```
 
+A description of each section is provided below.
+
+
 ### Header Section
 
 The header is a fixed-size section and has the following format (16 bytes):
+
 ```
 ----------------------------------------------------------------------
 | 0x55 0x4c 0x6f 0x67 0x01 0x12 0x35 | 0x01         | uint64_t       |
 | File magic (7B)                    | Version (1B) | Timestamp (8B) |
 ----------------------------------------------------------------------
 ```
 
-Version is the file format version, currently 1.
-Timestamp is a `uint64_t` integer, denotes the start of the logging in microseconds.
+- **File Magic (7 Bytes):** File type indicator that reads "ULogXYZ where XYZ is the magic bytes sequence `0x01 0x12 0x35`"
+- **Version (1 Byte):** File format version (currently 1)
+- **Timestamp (8 Bytes):** `uint64_t` integer that denotes when the logging started in microseconds.
+
+### Definition & Data Section Message Header
+
+The _Definitions and Data_ sections contain a number of **messages**. Each message is preceded by this header:
+
+```c
+struct message_header_s {
+  uint16_t msg_size;
+  uint8_t msg_type;
+};
+```
+
+- `msg_size` is the size of the message in bytes without the header.
+- `msg_type` defines the content, and is a single alphabet character.
+
+In practice, we include the `msg_size` and `msg_type` in each message definitions as the first 2 members, so the header `message_header_s` never gets used directly in most cases.
+
+:::note
+Messages described below are prefixed with the single alphabet that corresponds to it's `msg_type` in the beginning.
+:::
 
 
 ### Definitions Section
 
-Variable length section, contains version information, format definitions, and (initial) parameter values.
+The definitions section contains basic information such as software version, uORB Topic format, initial parameter values, and so on.
+
+The message types in this section are:
+1. [Flag Bits](#b-flag-bits-message)
+2. [Format Definition](#f-format-message)
+3. [Information](#i-information-message)
+4. [Multi Information](#m-multi-information-message)
+5. [Parameter](#p-parameter-message)
+6. [Default Parameter](#q-default-parameter-message)
+
+
+#### 'B': Flag Bits Message
+
+:::note
+This message must be the **first message** right after the header section, so that it has a fixed constant offset from the start of the file!
+:::
 
-The Definitions and Data sections consist of a stream of messages.
-Each starts with this header:
 ```c
-struct message_header_s {
```

Comment:
This was actually better for understanding, it makes things more clear and explicit. The fact that PX4 does it differently, isn't really relevant here.

Reply:
Gotcha 👌. One motive behind changing to this format was to expose the `msg_type` byte, and make it clear that each messages had their own type assigned 🤔. Will add it as a separate text!